### DISCUSSION TITLE
Rich Text Editor: Add Check Boxes

### DISCUSSION THREAD

**Discussion by kojid0** [element-hq/element-meta#998] [Ideas]
Similar to lists starting with bullet points or numbers, l'd like to see the opportunity to begin with clickable check boxes.
At first, the tick-nontick state doesn't need to be synced as it's probably harder to implement but it would be cool for a later release.

![124](https://user-images.githubusercontent.com/24477557/218970820-c5fa617e-99d7-4836-a738-6b499f43b759.png)

**Comment by peter-czy** [comment]
Nice idea, I know that the [Android Nextcloud Notes App](https://github.com/nextcloud/notes-android) has this feature. Maybe you could take a look at their code.

**Comment by Benkex** [comment]
So if I understand it correctly, in the first implementation of your suggestion you could check / uncheck them before sending and then they would freeze?

**Reply by kojid0** [reply]
No, you type a message with unchecked boxes. After you sent the message, you can click the boxes to check/uncheck them. 
It is probably quite some work to sync the check state across all users, so it would be fine if only the person who actively checks the boxes sees the check marks.

**Reply by t3chguy** [reply]
> It is probably quite some work to sync the check state across all users

Nope - you'd get it for free from message edits. Keep in mind checkboxes aren't allowed by the Matrix spec though. https://spec.matrix.org/v1.6/client-server-api/#mroommessage-msgtypes does not allow `input` tags

**Reply by kojid0** [reply]
@t3chguy Thank you. Could you please raise a short issue for me at the right place? :-)

**Reply by t3chguy** [reply]
I suggest https://github.com/matrix-org/matrix-spec

**Reply by eNkKVS4HxHc9jaq** [reply]
Altough it'd be nice to have interactice checkboxes, polls already exist for such scenarios. It being a design element sounds nice.
